### PR TITLE
[Merged by Bors] - feat: Lemmas specialised to `piFinset fun _ : Fin n ↦ s`

### PR DESCRIPTION
**Pull request by YaelDillies**
Those lemmas are specifically useful for rewriting backwards when the indexing type doesn't matter (and also sometimes useful forwardly).

From LeanAPAP

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Data/Fintype/BigOperators.lean [hunk 1]
```diff
@@ -112,9 +112,19 @@ namespace Fintype
 @[simp] lemma card_piFinset (s : ∀ i, Finset (α i)) :
     (piFinset s).card = ∏ i, (s i).card := by simp [piFinset, card_map]
 
+/-- This lemma is specifically designed to be used backwards, whence the specialisation to `Fin n`
+as the indexing type doesn't matter in practice. -/
```
Comment (b-mehta) [inline Mathlib/Data/Fintype/BigOperators.lean:116]:
```suggestion
as the indexing type doesn't matter in practice. The appropriate forward direction lemma here is
`card_piFinset`. -/
```
and below

File: Mathlib/Data/Fintype/Pi.lean [hunk 2]
```diff
@@ -56,6 +56,11 @@ theorem piFinset_empty [Nonempty α] : piFinset (fun _ => ∅ : ∀ i, Finset (
 lemma piFinset_nonempty : (piFinset s).Nonempty ↔ ∀ a, (s a).Nonempty := by
   simp [Finset.Nonempty, Classical.skolem]
 
+/-- This lemma is specifically designed to be used backwards, whence the specialisation to `Fin n`
```
Comment (b-mehta) [inline Mathlib/Data/Fintype/Pi.lean:59]: This comment doesn't really make sense here...

File: Mathlib/Data/Fintype/Pi.lean [hunk 3]
```diff
@@ -56,6 +56,9 @@ theorem piFinset_empty [Nonempty α] : piFinset (fun _ => ∅ : ∀ i, Finset (
 lemma piFinset_nonempty : (piFinset s).Nonempty ↔ ∀ a, (s a).Nonempty := by
   simp [Finset.Nonempty, Classical.skolem]
 
+lemma _root_.Finset.Nonempty.piFinset_const {s : Finset α} (hs : s.Nonempty) {ι : Type*} [Fintype ι]
```
Comment (b-mehta) [inline Mathlib/Data/Fintype/Pi.lean:59]: Let's put the Type argument and its instances before s and hs

**Review by b-mehta — APPROVED** [review]
bors d+

**Comment by YaelDillies** [comment]
bors merge

**Pull request closed without merging**